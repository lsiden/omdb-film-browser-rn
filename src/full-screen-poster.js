import React from "react"
import PropTypes from "prop-types"
import { Image, View } from "react-native"
import { connect } from "react-redux"

export const posterStyle = {
  flex: 1,
  height: undefined,
  width: undefined,
}

const wrapperStyle = {
  height: "100%",
  width: "100%",
  marginTop: 5,
  marginBottom: 5,
}

export const fullScreenPoster = ({ uri }) => {
  return (
    uri && (
      <View style={wrapperStyle}>
        <Image source={{ uri }} style={posterStyle} resizeMode={"contain"} />
      </View>
    )
  )
}

fullScreenPoster.propTypes = {
  uri: PropTypes.string.isRequired,
}

export default connect(state => ({
  uri: (state.poster || {}).uri,
}))(fullScreenPoster)
