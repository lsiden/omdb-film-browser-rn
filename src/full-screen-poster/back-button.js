import { connect } from "react-redux"

import { viewFilmDetails, updateIsFetching } from "actions"
import BackButton from "components/back-button"

const mapDispatchToProps = dispatch => ({
  dispatchToPrev: () => {
    dispatch(updateIsFetching(false))
    dispatch(viewFilmDetails())
  },
})

export default connect(null, mapDispatchToProps)(BackButton)
