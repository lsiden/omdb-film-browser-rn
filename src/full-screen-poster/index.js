import React from "react"
import PropTypes from "prop-types"
import { Image } from "react-native"
import { connect } from "react-redux"
import GestureRecognizer from "react-native-swipe-gestures"

import { fullScreenPosterStyles as style } from "styles"
import { viewFilmDetails, updateIsFetching } from "actions"

export const fullScreenPoster = ({ uri, dispatchToDetails }) => {
  return uri ? (
    <GestureRecognizer style={style.wrapperStyle} onSwipe={dispatchToDetails}>
      <Image
        source={{ uri }}
        style={style.posterStyle}
        resizeMode={"contain"}
        loadingIndicatorSource={require("assets/spinningwheel-300x216.gif")}
      />
    </GestureRecognizer>
  ) : null
}

fullScreenPoster.propTypes = {
  uri: PropTypes.string.isRequired,
  dispatchToDetails: PropTypes.func.isRequired,
}

export default connect(
  state => ({
    uri: (state.poster || {}).uri,
  }),
  dispatch => ({
    dispatchToDetails: () => {
      dispatch(updateIsFetching(false))
      dispatch(viewFilmDetails())
    },
  })
)(fullScreenPoster)
