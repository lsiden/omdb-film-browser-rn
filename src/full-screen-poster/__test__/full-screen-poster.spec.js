import React from "react"
import renderer from "react-native-test-utils"

import { fullScreenPoster as FullScreenPoster } from "full-screen-poster"

const defaultProps = () => {
  return {
    uri: "https://my.url.org",
    dispatchToDetails: jest.fn(),
  }
}

const createWrapper = (props = {}) =>
  renderer(<FullScreenPoster {...{ ...defaultProps(), ...props }} />)

it("renders", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
