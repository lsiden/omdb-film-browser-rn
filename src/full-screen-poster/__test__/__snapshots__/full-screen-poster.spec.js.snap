// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onSwipeLeft={[MockFunction]}
  onSwipeRight={[MockFunction]}
  style={
    Object {
      "height": "98%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginTop": 8,
      }
    }
  >
    <BackButton />
  </View>
  <Image
    loadingIndicatorSource={
      Object {
        "testUri": "../../../src/assets/spinningwheel-300x216.gif",
      }
    }
    resizeMode="contain"
    source={
      Object {
        "uri": "https://my.url.org",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": undefined,
        "marginBottom": 5,
        "marginTop": 5,
        "width": undefined,
      }
    }
  />
</View>
`;
